roles/load_aware_deploy_trimaran/tasks/main.yml

```
@@ -12,28 +12,55 @@
   delay: 3
   retries: 20
   # until all of the pods are in the running state
-  until: "'Completed' not in monitoring_enabled.stdout and 'Failed' not in monitoring_enabled.stdout and 'Pending' not in monitoring_enabled.stdout" 
+  until: "'Completed' not in monitoring_enabled.stdout and 'Failed' not in monitoring_enabled.stdout and 'Pending' not in monitoring_enabled.stdout"
 
-- name: Deploy Trimaran scheduler
+- name: Get Monitoring Secret
   shell:
+    oc get secret -n openshift-user-workload-monitoring
+      | grep  prometheus-user-workload-token
+      | head -n 1
+      | awk '{print $1 }'
+  register: monitoring_secret
+  delay: 3
+  retries: 20
+  until: "'prometheus-user-workload-token' in monitoring_secret.stdout"
+
+- name: Get Thanos Endpoint
+  shell:
+    oc get route thanos-querier -n openshift-monitoring -o json
+      | jq -r '.spec.host'
+  register: thanos_endpoint
+  delay: 3
+  retries: 20
+  until: "'thanos-querier' in thanos_endpoint.stdout"
+
+- name: Deploy Trimaran scheduler
+  shell: | 
     set -o errexit;
     set -o pipefail;
     set -o nounset;
     set -o errtrace;
+   
+    export THANOS_MONITORING_ENDPOINT={{ thanos_endpoint.stdout }}
 
-    export MONITORING_SECRET=$(oc get secret -n openshift-user-workload-monitoring
-      | grep  prometheus-user-workload-token
-      | head -n 1
-      | awk '{print $1 }'
-    )
+    export MONITORING_SECRET={{ monitoring_secret.stdout }}
 
-    export MONITORING_TOKEN=$(echo $(oc get secret $MONITORING_SECRET -n openshift-user-workload-monitoring -o json | jq -r '.data.token')
-      | base64 -d)
+    export MONITORING_TOKEN=$(echo $(oc get secret $MONITORING_SECRET -n openshift-user-workload-monitoring -o json | jq -r '.data.token') | base64 -d)
 
-    export THANOS_MONITORING_ENDPOINT=$(oc get route thanos-querier -n openshift-monitoring -o json
-      | jq -r '.spec.host')
+    # Dumping stderr here can result in leaking the token 
+    export TOKEN_SIZE=$(echo $MONITORING_TOKEN | wc -c 2> /dev/null)
 
-    cat {{ trimaran_setup_config }} | envsubst | oc apply -f -
+    if [[ TOKEN_SIZE -ge 1000 ]]; then
+      echo "DEPLOYING"  
+      cat {{ trimaran_setup_config }} | envsubst | oc apply -f -
+    else
+      echo "SKIPPING: token looks malformed"
+    fi
```

```suggestion
    echo "DEPLOYING"  
    cat {{ trimaran_setup_config }} | envsubst | oc apply -f -
```
your code would fail silently (=ignore the error you noticed), we never want that

could also be fixed with:
```
echo "ERROR: token looks malformed"
exit 1
```